MoreLinq/CountMethods.cs

```diff
@@ -0,0 +1,169 @@
+        /// <param name="source">The source sequence</param>
+        /// <param name="max">The maximun number of items a sequence must have for this
+        /// function to return true</param>
+        /// <exception cref="ArgumentNullException">source is null</exception>
+        /// <exception cref="ArgumentOutOfRangeException">max is negative</exception>
+        /// <returns><c>true</c> if the number of elements in the sequence is lesser than
+        /// or equal to the given integer or <c>false</c> otherwise.</returns>
+        /// <example>
+        /// <code>
+        /// var numbers = { 123, 456, 789 };
+        /// var result = numbers.AtMost(2);
+        /// </code>
+        /// The <c>result</c> variable will contain <c>false</c>.
+        /// </example>
+        public static bool AtMost<T>(this IEnumerable<T> source, int max)
+        {
+            if (max < 0) throw new ArgumentOutOfRangeException("max", "max must not be negative.");
+
+            return QuantityIterator(source, max + 1, count => count <= max);
+        }
+
+        /// <summary>
+        /// Returns true when the number of elements in the given sequence is greater than
+        /// or equal to the given integer.
+        /// This method throws an exception if the given integer is negative.
+        /// </summary>
+        /// <typeparam name="T">Element type of sequence</typeparam>
+        /// <param name="source">The source sequence</param>
+        /// <param name="length">The exactly number of items a sequence must have for this
+        /// function to return true</param>
+        /// <exception cref="ArgumentNullException">source is null</exception>
+        /// <exception cref="ArgumentOutOfRangeException">length is negative</exception>
+        /// <returns><c>true</c> if the number of elements in the sequence is equals
+        /// to the given integer or <c>false</c> otherwise.</returns>
+        /// <example>
+        /// <code>
+        /// var numbers = { 123, 456, 789 };
+        /// var result = numbers.Exactly(3);
+        /// </code>
+        /// The <c>result</c> variable will contain <c>true</c>.
+        /// </example>
+        public static bool Exactly<T>(this IEnumerable<T> source, int length)
+        {
+            if (length < 0) throw new ArgumentOutOfRangeException("length", "length must not be negative.");
+
+            return QuantityIterator(source, length + 1, count => count == length);
+        }
+
+        /// <summary>
+        /// Returns true when the number of elements in the given sequence is between (inclusive)
+        /// to the minimum and maximum given integer.
+        /// This method throws an exception if the minimum given integer is negative
+        /// or if the maximun given integer is lesser than the minimum integer.
+        /// </summary>
+        /// <remarks>
+        /// The number of items streamed will be greater than or equal to the given integer.
```

Comment:
Which integer? There are two of them. And you mean the number of items that will be _consumed_ rather than being _streamed out_? Generally streaming implies passing through so it causes word confusion here. My hunch is that you want to indicate how much of the source is read or consumed by this method.